File: store_admin_fe/src/shared/services/user.service.js
import client from '../api_handler/http-client'

const changePassword =  (data) => {
   return  client.request({
        url: '/user/change-password',
        method: 'post',
        data: data
    })
}

const createUser =  (data) => {
    return  client.request({
        url: '/user/create',
        method: 'post',
        data: data
    })
}

const updateUser =  (data) => {
    return  client.request({
        url: '/user/update',
        method: 'post',
        data: data
    })
}
const changeStatusUser =  (data) => {
    return  client.request({
        url: '/user/update-status',
        method: 'post',
        data: data
    })
}

const getUserById =  (request) => {
    return  client.request({
        url: '/user/get-by-id?id=' + request.id,
        method: 'get',
    })
}

const resetPassword =  (user) => {
    return client.request({
        url: '/user/reset-password',
        method: 'post',
        data: user
    })
}

const searchUser =  (data) => {
    return  client.request({
        url: '/user/search',
        method: 'post',
        data: data
    })
}

export {
    changePassword,
    getUserById,
    resetPassword,
    searchUser,
    createUser,
    updateUser,
    changeStatusUser
}

File: store_admin/src/main/java/vn/free/register/service/UserService.java
package vn.free.register.service;

import vn.free.register.request.user.UserSearch;
import vn.free.register.request.user.UserRQ;
import vn.free.register.response.ActionRes;
import vn.free.register.response.ResponseDTO;

public interface UserService {

    ResponseDTO searchUser(UserSearch search);
    ResponseDTO getUserById(UserRQ userRQ);
    ActionRes createUser(UserRQ userRQ);
    ActionRes updateUser(UserRQ userRQ);
    ActionRes updateStatusUser(UserRQ userRQ);
}
